js/sales-app.js

```diff
@@ -1,54 +1,185 @@
 'use strict';
 
-var hours = ['6am: ', '7am: ', '8am: ', '9am: ', '10am: ', '11am: ', '12pm: ', '1pm: ', '2pm: ', '3pm: ', '4pm: ', '5pm: ', '6pm: ', '7pm: ', '8pm: '];
+var hours = ['6am: ', '7am: ', '8am: ', '9am: ', '10am: ', '11am: ', '12pm: ', '1pm: ', '2pm: ', '3pm: ', '4pm: ', '5pm: ', '6pm: ', '7pm: ', '8pm: ', 'Total: '];
 
 var pike = {
-  min: 23,
-  max: 65,
-  avg: 6.3,
-  totalCookies: 'A bunch of ',
+  minCustomers: 23,
+  maxCustomers: 65,
+  avgCookiesPerCustomer: 6.3,
 
-  hourlyCookies: function() {
-    return ((Math.floor((Math.random() * (this.max - this.min) + this.min) * this.avg)));
+  hourlyCookies: function() { //this generates a random number of customers per hour and multiplies it by the average number of cookies
+    return Math.round((Math.floor(Math.random() * (this.maxCustomers - this.minCustomers + 1) + this.minCustomers)) * this.avgCookiesPerCustomer) + ' cookies';
   },
 
-  /*totalCookiesCounter: function() {
-
-    for(var i = 0; i < hours.length; i++) {
-      this.totalCookies += this.hourlyCookies();
-      console.log('There are a total of' + this.totalCookies);
+  cookieSales: function () {
+    var salesByHour = [];
+    var totalCookies = 0;
+    for(var i = 0 ; i < hours.length ; i++) {
+      if(i === (hours.length - 1)) { //this is for last item in array
+        salesByHour[i] = totalCookies + ' cookies';
+      } else { //this runs until we get to last element
+        salesByHour[i] = this.hourlyCookies(); //this adds values to 'salesByHour'
```

Comment:
Technically this works, however it is better practice to use `.push` to add new items to an array rather than setting a value to an index beyond the end of the array. That is, you should use `salesByHour.push(this.hourlyCookies())` instead.